src/detect-engine-port.c

```diff
@@ -1418,14 +1419,24 @@ DetectPort *PortParse(const char *str)
         port2[0] = '\0';
         port2++;
 
-        if(DetectPortIsValidRange(port))
-            dp->port = atoi(port);
-        else
-            goto error;
+        if (strcmp(port, "") != 0) {
+            if (DetectPortIsValidRange(port)) {
+                if (ByteExtractStringUint16(&dp->port, 10, 0,
```

Comment:
`DetectPortIsValidRange` and `ByteExtractStringUint16` are redundant
You could rather modify `DetectPortIsValidRange` to use `ByteExtractStringUint16` rather than `strtol`

Reply:
Makes sense.